File: _정답코드/3주차/05_배열_병합하기/Solution.java
import java.util.*;

public class Solution {
    public int[] solution(int[] arr1, int[] arr2) {
        int len1 = arr1.length;
        int len2 = arr2.length;
        int idx1 = 0, idx2 = 0;

        List<Integer> answer = new ArrayList<>();
        
        while(idx1 < len1 && idx2 < len2){
            if(arr1[idx1] < arr2[idx2]){
                answer.add(arr1[idx1]);
                idx1 ++;
            } else {
                answer.add(arr2[idx2]);
                idx2 ++;
            }
        }

        while(idx1 < len1){
            answer.add(arr1[idx1]);
            idx1 ++;
        }

        while(idx2 < len2){
            answer.add(arr2[idx2]);
            idx2 ++;
        }

        return answer.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: _정답코드/3주차/05_배열_병합하기/solution.py
def solution(arr1, arr2):
    len1 = len(arr1)
    len2 = len(arr2)
    idx1, idx2 = 0, 0
    answer = []

    while idx1 < len1 and idx2 < len2:
        if arr1[idx1] < arr2[idx2]:
            answer.append(arr1[idx1])
            idx1 += 1
        else:
            answer.append(arr2[idx2])
            idx2 += 1

    while idx1 < len1:
        answer.append(arr1[idx1])
        idx1 += 1

    while idx2 < len2:
        answer.append(arr2[idx2])
        idx2 += 1

    return answer

